index.js

```diff
@@ -32,7 +37,9 @@ module.exports = (input, opts) => new Promise((resolve, reject) => {
 				} else if (err instanceof TypeError) {
 					operation.stop();
 					reject(err);
-				} else if (!operation.retry(err)) {
+				} else if (operation.retry(err)) {
+					opts.onFailedAttempt(err, attemptNo, attemptsLeft);
```

Comment:
Decorating `err.attempt = attempt` and calling `opts.onFailedAttempt(err)` feels like a more natural API for node callbacks.

Reply:
Agreed.